tracer/src/Datadog.Trace/Iast/VulnerabilityBatch.cs

```diff
@@ -74,32 +78,40 @@ public void Add(Vulnerability vulnerability)
 
     public string ToJson()
     {
-        if (_vulnerabilitiesJson == null)
+        try
         {
-            if (_evidenceRedactor == null)
-            {
-                _vulnerabilitiesJson = JsonConvert.SerializeObject(this, Formatting.Indented, _serializerSettings);
-            }
-            else
+            if (_vulnerabilitiesJson == null)
             {
-                if (Sources != null)
+                if (_evidenceRedactor == null)
+                {
+                    _vulnerabilitiesJson = JsonConvert.SerializeObject(this, Formatting.Indented, _serializerSettings);
+                }
+                else
                 {
-                    foreach (var source in Sources)
+                    if (Sources != null)
                     {
-                        _evidenceRedactor.Process(source);
+                        foreach (var source in Sources)
+                        {
+                            _evidenceRedactor.Process(source);
+                        }
                     }
-                }
 
-                for (int x = 0; x < Vulnerabilities.Count; x++)
-                {
-                    Vulnerabilities[x] = _evidenceRedactor.RedactVulnerability(Vulnerabilities[x]);
-                }
+                    for (int x = 0; x < Vulnerabilities.Count; x++)
+                    {
+                        Vulnerabilities[x] = _evidenceRedactor.RedactVulnerability(Vulnerabilities[x]);
+                    }
 
-                _vulnerabilitiesJson = JsonConvert.SerializeObject(this, Formatting.Indented, _serializerSettings);
+                    _vulnerabilitiesJson = JsonConvert.SerializeObject(this, Formatting.Indented, _serializerSettings);
+                }
             }
-        }
 
-        return _vulnerabilitiesJson;
+            return _vulnerabilitiesJson;
+        }
+        catch (Exception ex)
+        {
+            Log.Error(ex, $"Error in {nameof(VulnerabilityBatch)}.{nameof(ToJson)}");
```

Comment:
Just wondering when this might throw? We need to be careful using `Log.Error` if it could fail in "normal" execution, because it can flood the logs (and also flood our telemetry)

Reply:
There should be no exceptions in normal execution inside this code. If we have an exception here, we should take a look because there will probably be a bug or unexpected situation.